File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

void DFS(vector<vector<int>> &computers, vector<bool> &visited, int node, int n){
    visited[node] = true;
    for(int i=0; i<n; i++){
        if(visited[i] == false && computers[node][i])
            DFS(computers, visited, i, n);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    vector<bool> visited (n, false);
    for(int i=0; i<n; i++)
        if(visited[i] == false){
            DFS(computers, visited, i, n);
            answer++;
        }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def BFS(n, computers, visited, start):
    visited[start] = True
    q = deque()
    q.append(start)
    
    while q:
        now = q.popleft()
        
        for i in range(n):
            if visited[i] or computers[now][i]==0:
                continue
            
            visited[i] = True
            q.append(i)
    
def solution(n, computers):
    answer = 0
    
    visited = [False] * n
    for i in range(n):
        if visited[i] == False:
            BFS(n, computers, visited, i)
            answer += 1;
    
    return answer